### ISSUE TITLE
Updating Regions Component with State Change

### ISSUE THREAD

**Issue by briankeane** [mspae/react-wavesurfer#41]
I am having trouble provoking a re-rendering of the regions when the props change.  Here is a simple example:

```
var React = require('react');
import WaveSurfer from 'react-wavesurfer';
window.WaveSurfer = require('wavesurfer.js');
let Regions = require('./Regions').default;

class SongCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      playing: false,
      pos: 0,
      regions: {
        eoi: {
          id: 'eoi',
          start: 5,
          end: 6
        }
      }
    };
    this.togglePlay = this.togglePlay.bind(this);
    this.handleTogglePlayClick = this.handleTogglePlayClick.bind(this);
    this.handlePosChange = this.handlePosChange.bind(this);
    this.handleMoveRegionsClick = this.handleMoveRegionsClick.bind(this);
  }
  togglePlay() {
    this.setState({
      playing: !this.state.playing
    });
  }
  handleTogglePlayClick() {
    this.togglePlay();
  }
  handlePosChange({ wavesurfer, originalArgs}) {
    this.setState({
      pos: originalArgs[0]
    })
  }
  handleMoveRegionsClick() {
    const newState = Object.assign({}, this.state, {
      regions: {
        eoi: {
          start: this.state.regions.eoi.start+10,
          end: this.state.regions.eoi.end+10,
          id: 'eoi'
        }
      }
    });
    this.setState(newState);
  }
  render () {
    return (
        <div>
          <div>
            <button type="button" className="btn btn-primary btn-xs" onClick={this.handleMoveRegionsClick}>Change Region</button>
          </div>
          <WaveSurfer
            audioFile={'https://ia902606.us.archive.org/35/items/shortpoetry_047_librivox/song_cjrg_teasdale_64kb.mp3'}
            pos={this.state.pos}
            onPosChange={this.handlePosChange}
            playing={this.state.playing}
          >
            <Regions regions={this.state.regions} />
          </WaveSurfer>
        </div>
      );
  }
};

export default SongCard;
```

When I call setState() within handleMoveRegionsClick(), SongCard's state is updated, but it does not trigger a reload of the Region's props.

Could anyone point me in the right direction for how to solve this?  Thanks!

**Comment by briankeane** [comment]
I modified the example app and got the same behavior here: https://github.com/briankeane/react-wavesurfer.git

in src/plugins/regions.js it does not call componentWillReceiveProps on state change.  If you have a sec to give me some guidance I'd really appreciate it.  Thanks!

**Comment by briankeane** [comment]
Finally got it working by changing shouldUpdateComponent() in src/react-wavesurfer.js to return true if the regions have been updated.

**Comment by zschmid** [comment]
@briankeane ,

Would you be able to post a code snippet here or a PR of the change you made? 
I'm having the exact same issue, changing:
```
  shouldComponentUpdate() {
    return true;
  }

```
in src/react-wavesurfer.js doesn't seem to resolve the issue for me. componentWillReceiveProps inside of the regions component is still not called.

**Comment by briankeane** [comment]
No prob -- I just made a new component called CustomWavesurfer.jsx and put this in it:

```
import Wavesurfer  from 'react-wavesurfer';

class CustomWavesurfer extends Wavesurfer {
  shouldComponentUpdate(nextProps, nextState) {
    return true
  }
}
export default CustomWavesurfer;
```

Then require my CustomWavesurfer instead of the real one.  I didn't pull request because I'm assuming there's a good reason for the way it's set up now.

**Comment by mspae** [comment]
Actually I think this functionality is just missing. The fix you posted rerenders very often (everytime a prop changes) and so is only a quickfix. The proper solution would be to add the `componentWillReceiveProps` function in the regions plugin and update the regions state. I'm open for a PR if anyone is up to it :balloon:

**Comment by mspae** [comment]
Should be fixed in v0.8.5 – please update.
